### ISSUE TITLE
Raw docstring (leading `r`) defeats form feed `\f` truncation

### ISSUE THREAD

**Issue by Kludex** [fastapi/fastapi#10998]
### Discussed in https://github.com/tiangolo/fastapi/discussions/10531

<div type='discussions-op-text'>

<sup>Originally posted by **jamesbraza** October 26, 2023</sup>
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.post("/lof")
def foo(arg: int = 5) -> int:
    """
    Some function.
    
    \f
    
    Args:
        arg: Some argument

    Returns:
        Some integer.
    """
    return arg
```

### Description

Running `ruff==0.1.3` on this, [`D301`](https://docs.astral.sh/ruff/rules/escape-sequence-in-docstring/) will autofix to make the docstring be a raw string (lead by `r`).

```python
@app.post("/lof")
def foo(arg: int = 5) -> int:
    r"""
    Some function.
    
    \f
    
    Args:
        arg: Some argument

    Returns:
        Some integer.
    """
    return arg
```

However, this breaks the form feed `\f` truncation from working, per https://fastapi.tiangolo.com/advanced/path-operation-advanced-configuration/#advanced-description-from-docstring

I think form feed `\f` truncation should still work even if the docstring is a raw string.

### Operating System

macOS

### Operating System Details

n/a

### FastAPI Version

0.104.0

### Pydantic Version

2.4.2

### Python Version

3.11.5

### Additional Context

_No response_</div>

**Comment by KaniKim** [comment]
I little bit handled this by adding some code.
I don't know what you actually want, but you want like this code situation?

![fast](https://github.com/tiangolo/fastapi/assets/19832624/37c7d54a-76ee-4af6-91d1-5a2587a523bc)

Above code is with raw string that print to normal string.
And also here is my code changed version.

![fastapi-code](https://github.com/tiangolo/fastapi/assets/19832624/38765c9b-b7d1-44d7-99a9-da2f7a135a05)

Here is code about I write

![carbon](https://github.com/tiangolo/fastapi/assets/19832624/abffea39-12bc-49b2-a611-fe92dc0841b1)

**Comment by dolfinus** [comment]
Well, `r"\f"` ==`"\\f"` != `"\f"`

With change you propose to docstrings like `This endpoint saves data to C:\files\data` will be truncated to `This endpoint saves data to C:`

**Comment by ShreySinha02** [comment]
opened the #11149 pr 
used codecs to convert the raw string to normal string
'''python
  self.description = codecs.decode(self.description, "unicode_escape")
'''

**Comment by YuriiMotov** [comment]
I think the best way is to add a note in docs that `r"\f"` is not equal to `"\f"` and if one wants to use [truncating by form feed character](https://fastapi.tiangolo.com/advanced/path-operation-advanced-configuration/#advanced-description-from-docstring) they shouldn't use raw string and should disable ruff rule for this line.

**Comment by tiangolo** [comment]
Yep, agreed with @YuriiMotov, `r"\f"` and `"\f"` are two different things.

The linter rule is assuming that you are *not* intending to use a form feed, but in this case, you are. So the rule here is wrong, if you want to use the form feed character, you should disable that rule for that line.

---

Additionally, the idea of the `\f` form feed character is already a bit weird to me, I never use it.

I have never seen a need to separate the docs of a path operation function and some additional docstring part of it that is in some way not part of the docstring.

I would actually suggest not using that, but anyway, some people want it, so we'll keep the functionality there, but it wouldn't make sense to have some extra custom behavior when the rule is the one that is being a bit weird here.

Given that, I'll pass on this feature request, but thanks for the interest everyone! :coffee:
